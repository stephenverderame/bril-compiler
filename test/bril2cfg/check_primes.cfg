digraph {
	subgraph cluster_main {
		label="main(n: int)";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="v2 := 1\ni := v2", shape="rectangle", style="rounded"];
		main_3 [label="v3 := i\nv4 := n\nv5 := v3 < v4\nif v5", shape="rectangle", style="rounded"];
		main_4 [label="v6 := i\nisPrime := checkPrime(v6)\nisPrime := isPrime\nv8 := isPrime\nif v8", shape="rectangle", style="rounded"];
		main_5 [label="v9 := 1\nprint v9\nv10 := 0", shape="rectangle", style="rounded"];
		main_6 [label="v11 := 0\nprint v11\nv12 := 0", shape="rectangle", style="rounded"];
		main_7 [label="v13 := i\nv14 := 1\nv15 := v13 + v14\ni := v15", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4 [label="T"];
		main_4 -> main_5 [label="T"];
		main_4 -> main_6 [label="F"];
		main_5 -> main_7;
		main_6 -> main_7;
		main_7 -> main_3;
	}
	subgraph cluster_checkPrime {
		label="checkPrime(x: int)";
		rankdir="TB";
		checkPrime_0 [label="START", shape="rectangle", style="rounded"];
		checkPrime_2 [label="v1 := x\nv2 := 1\nv3 := v1 <= v2\nif v3", shape="rectangle", style="rounded"];
		checkPrime_3 [label="v4 := false\nreturn v4", shape="rectangle", style="rounded"];
		checkPrime_4 [label="v6 := 2\ni := v6", shape="rectangle", style="rounded"];
		checkPrime_5 [label="v7 := i\nv8 := x\nv9 := v7 < v8\nif v9", shape="rectangle", style="rounded"];
		checkPrime_6 [label="v10 := x\nv11 := i\nv12 := v10 / v11\ndiv := v12\nv13 := div\nv14 := i\nv15 := v13 * v14\nv16 := x\nv17 := v15 == v16\nisDivisible := v17\nv19 := isDivisible\nif v19", shape="rectangle", style="rounded"];
		checkPrime_7 [label="v20 := false\nreturn v20", shape="rectangle", style="rounded"];
		checkPrime_8 [label="v21 := i\nv22 := 1\nv23 := v21 + v22\ni := v23", shape="rectangle", style="rounded"];
		checkPrime_9 [label="v24 := true\nreturn v24", shape="rectangle", style="rounded"];

		checkPrime_0 -> checkPrime_2;
		checkPrime_2 -> checkPrime_3 [label="T"];
		checkPrime_2 -> checkPrime_4 [label="F"];
		checkPrime_4 -> checkPrime_5;
		checkPrime_5 -> checkPrime_6 [label="T"];
		checkPrime_5 -> checkPrime_9 [label="F"];
		checkPrime_6 -> checkPrime_7 [label="T"];
		checkPrime_6 -> checkPrime_8 [label="F"];
		checkPrime_8 -> checkPrime_5;
	}
}
