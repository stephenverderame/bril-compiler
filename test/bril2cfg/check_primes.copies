digraph {
	subgraph cluster_main {
		label="main(n: int)";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[]\n----------\nv2 := 1\ni := v2\n----------\n[i = v2]", shape="rectangle", style="rounded"];
		main_3 [label="[isPrime = isPrime, v4 = n, v8 = isPrime]\n----------\nv3 := i\nv4 := n\nv5 := v3 < v4\nif v5\n----------\n[isPrime = isPrime, v3 = i, v4 = n, v8 = isPrime]", shape="rectangle", style="rounded"];
		main_4 [label="[isPrime = isPrime, v3 = i, v4 = n, v8 = isPrime]\n----------\nv6 := i\nisPrime := checkPrime(v6)\nisPrime := isPrime\nv8 := isPrime\nif v8\n----------\n[isPrime = isPrime, v3 = i, v4 = n, v6 = i, v8 = isPrime]", shape="rectangle", style="rounded"];
		main_5 [label="[isPrime = isPrime, v3 = i, v4 = n, v6 = i, v8 = isPrime]\n----------\nv9 := 1\nprint v9\nv10 := 0\n----------\n[isPrime = isPrime, v3 = i, v4 = n, v6 = i, v8 = isPrime]", shape="rectangle", style="rounded"];
		main_6 [label="[isPrime = isPrime, v3 = i, v4 = n, v6 = i, v8 = isPrime]\n----------\nv11 := 0\nprint v11\nv12 := 0\n----------\n[isPrime = isPrime, v3 = i, v4 = n, v6 = i, v8 = isPrime]", shape="rectangle", style="rounded"];
		main_7 [label="[isPrime = isPrime, v3 = i, v4 = n, v6 = i, v8 = isPrime]\n----------\nv13 := i\nv14 := 1\nv15 := v13 + v14\ni := v15\n----------\n[i = v15, isPrime = isPrime, v4 = n, v8 = isPrime]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4 [label="T"];
		main_4 -> main_5 [label="T"];
		main_4 -> main_6 [label="F"];
		main_5 -> main_7;
		main_6 -> main_7;
		main_7 -> main_3;
	}
	subgraph cluster_checkPrime {
		label="checkPrime(x: int)";
		rankdir="TB";
		checkPrime_0 [label="START", shape="rectangle", style="rounded"];
		checkPrime_2 [label="[]\n----------\nv1 := x\nv2 := 1\nv3 := v1 <= v2\nif v3\n----------\n[v1 = x]", shape="rectangle", style="rounded"];
		checkPrime_3 [label="[v1 = x]\n----------\nv4 := false\nreturn v4\n----------\n[v1 = x]", shape="rectangle", style="rounded"];
		checkPrime_4 [label="[v1 = x]\n----------\nv6 := 2\ni := v6\n----------\n[i = v6, v1 = x]", shape="rectangle", style="rounded"];
		checkPrime_5 [label="[div = v12, isDivisible = v17, v1 = x, v10 = x, v13 = div, v16 = x, v19 = isDivisible, v8 = x]\n----------\nv7 := i\nv8 := x\nv9 := v7 < v8\nif v9\n----------\n[div = v12, isDivisible = v17, v1 = x, v10 = x, v13 = div, v16 = x, v19 = isDivisible, v7 = i, v8 = x]", shape="rectangle", style="rounded"];
		checkPrime_6 [label="[div = v12, isDivisible = v17, v1 = x, v10 = x, v13 = div, v16 = x, v19 = isDivisible, v7 = i, v8 = x]\n----------\nv10 := x\nv11 := i\nv12 := v10 / v11\ndiv := v12\nv13 := div\nv14 := i\nv15 := v13 * v14\nv16 := x\nv17 := v15 == v16\nisDivisible := v17\nv19 := isDivisible\nif v19\n----------\n[div = v12, isDivisible = v17, v1 = x, v10 = x, v11 = i, v13 = div, v14 = i, v16 = x, v19 = isDivisible, v7 = i, v8 = x]", shape="rectangle", style="rounded"];
		checkPrime_7 [label="[div = v12, isDivisible = v17, v1 = x, v10 = x, v11 = i, v13 = div, v14 = i, v16 = x, v19 = isDivisible, v7 = i, v8 = x]\n----------\nv20 := false\nreturn v20\n----------\n[div = v12, isDivisible = v17, v1 = x, v10 = x, v11 = i, v13 = div, v14 = i, v16 = x, v19 = isDivisible, v7 = i, v8 = x]", shape="rectangle", style="rounded"];
		checkPrime_8 [label="[div = v12, isDivisible = v17, v1 = x, v10 = x, v11 = i, v13 = div, v14 = i, v16 = x, v19 = isDivisible, v7 = i, v8 = x]\n----------\nv21 := i\nv22 := 1\nv23 := v21 + v22\ni := v23\n----------\n[div = v12, i = v23, isDivisible = v17, v1 = x, v10 = x, v13 = div, v16 = x, v19 = isDivisible, v8 = x]", shape="rectangle", style="rounded"];
		checkPrime_9 [label="[div = v12, isDivisible = v17, v1 = x, v10 = x, v13 = div, v16 = x, v19 = isDivisible, v7 = i, v8 = x]\n----------\nv24 := true\nreturn v24\n----------\n[div = v12, isDivisible = v17, v1 = x, v10 = x, v13 = div, v16 = x, v19 = isDivisible, v7 = i, v8 = x]", shape="rectangle", style="rounded"];

		checkPrime_0 -> checkPrime_2;
		checkPrime_2 -> checkPrime_3 [label="T"];
		checkPrime_2 -> checkPrime_4 [label="F"];
		checkPrime_4 -> checkPrime_5;
		checkPrime_5 -> checkPrime_6 [label="T"];
		checkPrime_5 -> checkPrime_9 [label="F"];
		checkPrime_6 -> checkPrime_7 [label="T"];
		checkPrime_6 -> checkPrime_8 [label="F"];
		checkPrime_8 -> checkPrime_5;
	}
}
