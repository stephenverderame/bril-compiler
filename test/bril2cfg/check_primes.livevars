digraph {
	subgraph cluster_main {
		label="main(n: int)";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[n]\n----------\nv2 := 1\ni := v2\n----------\n[i, n]", shape="rectangle", style="rounded"];
		main_3 [label="[i, n]\n----------\nv3 := i\nv4 := n\nv5 := v3 < v4\nif v5\n----------\n[i, n]", shape="rectangle", style="rounded"];
		main_4 [label="[i, n]\n----------\nv6 := i\nisPrime := checkPrime(v6)\nisPrime := isPrime\nv8 := isPrime\nif v8\n----------\n[i, n]", shape="rectangle", style="rounded"];
		main_5 [label="[i, n]\n----------\nv9 := 1\nprint v9\nv10 := 0\n----------\n[i, n]", shape="rectangle", style="rounded"];
		main_6 [label="[i, n]\n----------\nv11 := 0\nprint v11\nv12 := 0\n----------\n[i, n]", shape="rectangle", style="rounded"];
		main_7 [label="[i, n]\n----------\nv13 := i\nv14 := 1\nv15 := v13 + v14\ni := v15\n----------\n[i, n]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4 [label="T"];
		main_4 -> main_5 [label="T"];
		main_4 -> main_6 [label="F"];
		main_5 -> main_7;
		main_6 -> main_7;
		main_7 -> main_3;
	}
	subgraph cluster_checkPrime {
		label="checkPrime(x: int)";
		rankdir="TB";
		checkPrime_0 [label="START", shape="rectangle", style="rounded"];
		checkPrime_2 [label="[x]\n----------\nv1 := x\nv2 := 1\nv3 := v1 <= v2\nif v3\n----------\n[x]", shape="rectangle", style="rounded"];
		checkPrime_3 [label="[]\n----------\nv4 := false\nreturn v4\n----------\n[]", shape="rectangle", style="rounded"];
		checkPrime_4 [label="[x]\n----------\nv6 := 2\ni := v6\n----------\n[i, x]", shape="rectangle", style="rounded"];
		checkPrime_5 [label="[i, x]\n----------\nv7 := i\nv8 := x\nv9 := v7 < v8\nif v9\n----------\n[i, x]", shape="rectangle", style="rounded"];
		checkPrime_6 [label="[i, x]\n----------\nv10 := x\nv11 := i\nv12 := v10 / v11\ndiv := v12\nv13 := div\nv14 := i\nv15 := v13 * v14\nv16 := x\nv17 := v15 == v16\nisDivisible := v17\nv19 := isDivisible\nif v19\n----------\n[i, x]", shape="rectangle", style="rounded"];
		checkPrime_7 [label="[]\n----------\nv20 := false\nreturn v20\n----------\n[]", shape="rectangle", style="rounded"];
		checkPrime_8 [label="[i, x]\n----------\nv21 := i\nv22 := 1\nv23 := v21 + v22\ni := v23\n----------\n[i, x]", shape="rectangle", style="rounded"];
		checkPrime_9 [label="[]\n----------\nv24 := true\nreturn v24\n----------\n[]", shape="rectangle", style="rounded"];

		checkPrime_0 -> checkPrime_2;
		checkPrime_2 -> checkPrime_3 [label="T"];
		checkPrime_2 -> checkPrime_4 [label="F"];
		checkPrime_4 -> checkPrime_5;
		checkPrime_5 -> checkPrime_6 [label="T"];
		checkPrime_5 -> checkPrime_9 [label="F"];
		checkPrime_6 -> checkPrime_7 [label="T"];
		checkPrime_6 -> checkPrime_8 [label="F"];
		checkPrime_8 -> checkPrime_5;
	}
}
