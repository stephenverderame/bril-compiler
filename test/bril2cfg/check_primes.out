digraph {
	subgraph cluster_main {
		label="main";
		rankdir="TB";
		main_0 [label="START", shape="ellipse"];
		main_2 [label="v2 := 1", shape="ellipse"];
		main_3 [label="i := v2", shape="ellipse"];
		main_4 [label="v3 := i", shape="ellipse"];
		main_5 [label="v4 := n", shape="ellipse"];
		main_6 [label="v5 := v3 < v4", shape="ellipse"];
		main_7 [label="if v5", shape="ellipse"];
		main_8 [label="v6 := i", shape="ellipse"];
		main_9 [label="isPrime := checkPrime(v6)", shape="ellipse"];
		main_10 [label="isPrime := isPrime", shape="ellipse"];
		main_11 [label="v8 := isPrime", shape="ellipse"];
		main_12 [label="if v8", shape="ellipse"];
		main_13 [label="v9 := 1", shape="ellipse"];
		main_14 [label="print v9", shape="ellipse"];
		main_15 [label="v10 := 0", shape="ellipse"];
		main_17 [label="v11 := 0", shape="ellipse"];
		main_18 [label="print v11", shape="ellipse"];
		main_19 [label="v12 := 0", shape="ellipse"];
		main_20 [label="v13 := i", shape="ellipse"];
		main_21 [label="v14 := 1", shape="ellipse"];
		main_22 [label="v15 := v13 + v14", shape="ellipse"];
		main_23 [label="i := v15", shape="ellipse"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4;
		main_4 -> main_5;
		main_5 -> main_6;
		main_6 -> main_7;
		main_7 -> main_8 [label="T"];
		main_8 -> main_9;
		main_9 -> main_10;
		main_10 -> main_11;
		main_11 -> main_12;
		main_12 -> main_13 [label="T"];
		main_12 -> main_17 [label="F"];
		main_13 -> main_14;
		main_14 -> main_15;
		main_15 -> main_20;
		main_17 -> main_18;
		main_18 -> main_19;
		main_19 -> main_20;
		main_20 -> main_21;
		main_21 -> main_22;
		main_22 -> main_23;
		main_23 -> main_4;
	}
	subgraph cluster_checkPrime {
		label="checkPrime";
		rankdir="TB";
		checkPrime_0 [label="START", shape="ellipse"];
		checkPrime_2 [label="v1 := x", shape="ellipse"];
		checkPrime_3 [label="v2 := 1", shape="ellipse"];
		checkPrime_4 [label="v3 := v1 <= v2", shape="ellipse"];
		checkPrime_5 [label="if v3", shape="ellipse"];
		checkPrime_6 [label="v4 := false", shape="ellipse"];
		checkPrime_7 [label="return v4", shape="ellipse"];
		checkPrime_8 [label="v6 := 2", shape="ellipse"];
		checkPrime_9 [label="i := v6", shape="ellipse"];
		checkPrime_10 [label="v7 := i", shape="ellipse"];
		checkPrime_11 [label="v8 := x", shape="ellipse"];
		checkPrime_12 [label="v9 := v7 < v8", shape="ellipse"];
		checkPrime_13 [label="if v9", shape="ellipse"];
		checkPrime_14 [label="v10 := x", shape="ellipse"];
		checkPrime_15 [label="v11 := i", shape="ellipse"];
		checkPrime_16 [label="v12 := v10 / v11", shape="ellipse"];
		checkPrime_17 [label="div := v12", shape="ellipse"];
		checkPrime_18 [label="v13 := div", shape="ellipse"];
		checkPrime_19 [label="v14 := i", shape="ellipse"];
		checkPrime_20 [label="v15 := v13 * v14", shape="ellipse"];
		checkPrime_21 [label="v16 := x", shape="ellipse"];
		checkPrime_22 [label="v17 := v15 == v16", shape="ellipse"];
		checkPrime_23 [label="isDivisible := v17", shape="ellipse"];
		checkPrime_24 [label="v19 := isDivisible", shape="ellipse"];
		checkPrime_25 [label="if v19", shape="ellipse"];
		checkPrime_26 [label="v20 := false", shape="ellipse"];
		checkPrime_27 [label="return v20", shape="ellipse"];
		checkPrime_28 [label="v21 := i", shape="ellipse"];
		checkPrime_29 [label="v22 := 1", shape="ellipse"];
		checkPrime_30 [label="v23 := v21 + v22", shape="ellipse"];
		checkPrime_31 [label="i := v23", shape="ellipse"];
		checkPrime_33 [label="v24 := true", shape="ellipse"];
		checkPrime_34 [label="return v24", shape="ellipse"];

		checkPrime_0 -> checkPrime_2;
		checkPrime_2 -> checkPrime_3;
		checkPrime_3 -> checkPrime_4;
		checkPrime_4 -> checkPrime_5;
		checkPrime_5 -> checkPrime_6 [label="T"];
		checkPrime_5 -> checkPrime_8 [label="F"];
		checkPrime_6 -> checkPrime_7;
		checkPrime_8 -> checkPrime_9;
		checkPrime_9 -> checkPrime_10;
		checkPrime_10 -> checkPrime_11;
		checkPrime_11 -> checkPrime_12;
		checkPrime_12 -> checkPrime_13;
		checkPrime_13 -> checkPrime_14 [label="T"];
		checkPrime_13 -> checkPrime_33 [label="F"];
		checkPrime_14 -> checkPrime_15;
		checkPrime_15 -> checkPrime_16;
		checkPrime_16 -> checkPrime_17;
		checkPrime_17 -> checkPrime_18;
		checkPrime_18 -> checkPrime_19;
		checkPrime_19 -> checkPrime_20;
		checkPrime_20 -> checkPrime_21;
		checkPrime_21 -> checkPrime_22;
		checkPrime_22 -> checkPrime_23;
		checkPrime_23 -> checkPrime_24;
		checkPrime_24 -> checkPrime_25;
		checkPrime_25 -> checkPrime_26 [label="T"];
		checkPrime_25 -> checkPrime_28 [label="F"];
		checkPrime_26 -> checkPrime_27;
		checkPrime_28 -> checkPrime_29;
		checkPrime_29 -> checkPrime_30;
		checkPrime_30 -> checkPrime_31;
		checkPrime_31 -> checkPrime_10;
		checkPrime_33 -> checkPrime_34;
	}
}
